packages/p2p-v2/src/pages/my-ads/screens/MyAds/MyAdsTable/MyAdsTable.tsx

```diff
@@ -46,13 +47,48 @@ const columns = [
     },
 ];
 
+type TMyAdsDisplayWrapperProps = {
+    isPaused: boolean;
+    onClickToggle: () => void;
+};
+
+const MyAdsDisplayWrapper = ({ children, isPaused, onClickToggle }: PropsWithChildren<TMyAdsDisplayWrapperProps>) => {
+    const { isMobile } = useDevice();
+    if (isMobile) {
+        return (
+            <FullPageMobileWrapper
+                renderFooter={() => (
+                    <Button isFullWidth size='lg' textSize='md'>
+                        Create new ad
+                    </Button>
+                )}
+                renderHeader={() => <MyAdsToggle isPaused={isPaused} onClickToggle={onClickToggle} />}
+                shouldShowBackIcon={false}
+            >
+                {children}
+            </FullPageMobileWrapper>
+        );
+    }
+
+    return (
+        <>
+            <div className='flex items-center justify-between my-[1.6rem]'>
+                <Button size='lg' textSize='sm'>
+                    Create new ad
+                </Button>
+                <MyAdsToggle isPaused={isPaused} onClickToggle={onClickToggle} />
+            </div>
+            {children}
+        </>
+    );
+};
+
```

Comment:
can we separate this into its on file? 👀 can just have it under the MyAdsTable Folder